src/main/java/eu/proteus/producer/tasks/ProteusStreamingTask.java

```diff
@@ -9,204 +20,200 @@
-        return (currentRow.getX() - this.model.getLastCoilRow().getX())
-                * (ProteusData.COIL_TIME / ProteusData.getXmax(currentRow.getCoilId()));
-    }
-
-    private void updateCoilTimestampStart() {
-        this.model.setLastCoilStart(new Date());
-    }
-
-    /**
-     * Updates the program status after each iteration
-     *
-     * @param row Current row
-     */
-    private void updateStatus(SensorMeasurement row) {
-        this.model.setLastCoilRow(row);
-    }
-
-    /**
-     * Produces a new message containing the current row. It uses
-     *
-     * @param row
-     */
-    private void produceMessage(SensorMeasurement row) {
-        logger.debug("Producing row: " + row);
-        ProteusKafkaProducer.produce(row);
-    }
-
-    private boolean discardWrongCoilRows(SensorMeasurement row) {
-        return row != null && row.getX() > 0 && row.getCoilId() > 0;
-    }
-
-    /**
-     * Apply a specific delay
-     *
-     * @param delay delay time
-     */
-    public void applyDelay(double delay) {
-        if(delay > 7000D) { //avoid to much logs
-            logger.debug("Sleeping " + this.getClass().getName()+" for " + delay + "ms");
-        }
-        try {
-            Thread.sleep((long) delay);
-        } catch (InterruptedException e) {
-            e.printStackTrace();
-        }
-        if(delay > 7000D) { //avoid to much logs
-            logger.debug(this.getClass().getName() + " is alive again");
-        }
-    }
-
+		}
+	}
+
+	/**
+	 * Calculates a delay time for the current row, based on the X value
+	 * difference between current and previous row
+	 *
+	 * @param currentRow
+	 *            current row
+	 * @return
+	 */
+	private double calculateDelayBetweenCurrentAndLastRow(SensorMeasurement currentRow) {
+		return (currentRow.getX() - this.model.getLastCoilRow().getX())
+				* (ProteusData.COIL_TIME / ProteusData.getXmax(currentRow.getCoilId()));
+	}
+
+	private void updateCoilTimestampStart() {
+		this.model.setLastCoilStart(new Date());
+	}
+
+	/**
+	 * Updates the program status after each iteration
+	 *
+	 * @param row
+	 *            Current row
+	 */
+	private void updateStatus(SensorMeasurement row) {
+		this.model.setLastCoilRow(row);
+	}
+
+	/**
+	 * Produces a new message containing the current row. It uses
+	 *
+	 * @param row
+	 */
+	private void produceMessage(SensorMeasurement row) {
+		logger.debug("Producing row: " + row);
+		ProteusKafkaProducer.produce(row);
+	}
+
+	private boolean discardWrongCoilRows(SensorMeasurement row) {
+		return row != null && row.getX() > 0 && row.getCoilId() > 0 && row.getCoilId() < 45 && row.getCoilId() != 27
+				&& row.getCoilId() != 9 && row.getCoilId() != 19 && row.getCoilId() != 20 && row.getCoilId() != 22
+				&& row.getCoilId() != 31 && row.getCoilId() != 32 && row.getCoilId() != 36 && row.getCoilId() != 38
+				&& row.getCoilId() != 39 && row.getCoilId() != 41;
```

Comment:
What we should ignore is records with a given var id, but you are using row.getCoilId instead of row.getVarId.

Furthermore, I recommend to create a local array containing all the var ids to ignore, and then loop them in the discardWrongCoils function.